testhelpers/image_privileges.go

```diff
@@ -0,0 +1,41 @@
+package testhelpers
+
+import (
+	"regexp"
+	"strings"
+)
+
+var urlRegex = regexp.MustCompile(`v2\/(.*)\/(?:blobs|manifests|tags)`)
+
+type ImagePrivileges struct {
+	readable bool
+	writable bool
+}
+
+// Creates a new ImagePrivileges, use "readable" or "writable" to set the properties accordingly.
+// For examples:
+// NewImagePrivileges("") returns ImagePrivileges{readable: false, writable: false}
+// NewImagePrivileges("foo-readable") returns ImagePrivileges{readable: true, writable: false}
+// NewImagePrivileges("foo-writable") returns ImagePrivileges{readable: false, writable: true}
+// NewImagePrivileges("foo-writable-readable") returns ImagePrivileges{readable: true, writable: true}
+func NewImagePrivileges(imageName string) (priv ImagePrivileges) {
+	priv.readable = strings.Contains(imageName, "readable")
+	priv.writable = strings.Contains(imageName, "writable")
+	return
+}
```

Comment:
Maybe I'm wrong but according to what I saw in our code, we usually don't set the returned variable name in the function definition. We define it as part of the function and return it explicitly.
If I'm right, maybe it's better to change the function to:

```suggestion
func NewImagePrivileges(imageName string) (priv ImagePrivileges) {
	image ImagePrivileges
	image.readable = strings.Contains(imageName, "readable")
	image.writable = strings.Contains(imageName, "writable")
	return image
}
```